File: 0605-can-place-flowers/0605-can-place-flowers.php
class Solution {

    /**
     * @param Integer[] $flowerbed
     * @param Integer $n
     * @return Boolean
     */
    function canPlaceFlowers($flowerbed, $n) {
        array_unshift($flowerbed, 0);
        array_push($flowerbed, 0);

        $flowerbedLength = count($flowerbed);
        for($i = 1; $i < $flowerbedLength - 1; $i++){
            if($flowerbed[$i] + $flowerbed[$i - 1] + $flowerbed[$i + 1] == 0){
                $n--;
                $i++;

            }
            if($n <= 0) return true;
        }
        
        return $n <= 0;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed, n):
        flowerbed.insert(0, 0)
        flowerbed.append(0)
        
        flowerbed_length = len(flowerbed)
        i = 1
        while i < flowerbed_length - 1:
            if flowerbed[i] + flowerbed[i - 1] + flowerbed[i + 1] == 0:
                n -= 1
                i += 1

            if n <= 0:
                return True
            
            i += 1
        
        return n <= 0
